### PR TITLE
Feature: Introduce the new NavigationViewItem style to Properties window

### PR DESCRIPTION
**Pull request by 0x5bfa**
### Description

Part of the renovation.

### Motivation and Context

- #11584

### Tasks

- [x] Change style
- [x] Change style to be able to display two different icon corresponding states
- [x] Collapse pane on smaller windows
- [x] Make default window size wider

### Validation

- [x] Built and ran the app
- [x] Tested the changes for accessibility

### Screenshots

Default mode
image

Compact mode
image

### CODE REVIEW

**Comment by onein528** [comment]
> Change style to be able to display two different icon corresponding states

I can do this primitively(I can do fast) like this:
```xaml
<StackPanel Orientation="Horizontal">
    <Grid>
        <OpacityIcon Visibility="{x:Bind IsSelected, Converter...}" /> <!--  Unselected  -->
        <OpacityIcon Visibility="{x:Bind IsSelected}" /> <!--  Selected  -->
    </Grid>

    <TextBlock Text="{x:Bind Name}" />
</StackPanel>
```

But Idk, maybe still adding VisualState is more better (have to add codes a lot, though).

**Comment by yaira2** [comment]
VisualStates are better and less xaml overall.

File: src/Files.App/Views/Properties/MainPropertiesPage.xaml [hunk 1]
```diff
@@ -33,49 +33,31 @@
...
 			x:Name="MainPropertiesWindowNavigationView"
 			Grid.Row="1"
-			Width="80"
 			IsBackButtonVisible="Collapsed"
 			IsPaneToggleButtonVisible="False"
 			IsSettingsVisible="False"
 			IsTitleBarAutoPaddingEnabled="False"
 			MenuItemsSource="{x:Bind NavViewItems, Mode=TwoWay}"
-			OpenPaneLength="88"
+			OpenPaneLength="200"
```
Comment (yaira2) [inline src/Files.App/Views/Properties/MainPropertiesPage.xaml:60]: Is this what we used in settings?

Comment (onein528) [reply src/Files.App/Views/Properties/MainPropertiesPage.xaml:60]: 260 in settings. should be the same?

Comment (yaira2) [reply src/Files.App/Views/Properties/MainPropertiesPage.xaml:60]: We can try with 200 👍

**Comment by 0x5bfa** [comment]
Do you @yair100 me try that in here PR?

**Comment by yaira2** [comment]
I already implemented it in propeties but it's missing the new styles

File: src/Files.App/Views/Properties/MainPropertiesPage.xaml [hunk 2]
```diff
@@ -37,127 +38,104 @@
...
 
 								<!--  Icon  -->
 								<Viewbox
-									Width="24"
-									Height="24"
+									Width="16"
+									Height="16"
 									VerticalAlignment="Center">
-									<usercontrols:OpacityIcon IsSelected="{x:Bind IsSelected, Mode=OneWay}" Style="{x:Bind OpacityIconStyle, Mode=OneWay}" />
+									<uc:OpacityIcon IsSelected="{x:Bind IsSelected, Mode=OneWay}" Style="{x:Bind OpacityIconStyle}" />
```
Comment (yaira2) [inline src/Files.App/Views/Properties/MainPropertiesPage.xaml:79]:
```suggestion
									<uc:OpacityIcon IsSelected="{x:Bind IsSelected, Mode=OneWay}" Style="{x:Bind OpacityIconStyle, Mode=OneWay}" />
```

Comment (yaira2) [reply src/Files.App/Views/Properties/MainPropertiesPage.xaml:79]: We always define mode

Comment (onein528) [reply src/Files.App/Views/Properties/MainPropertiesPage.xaml:79]: Can you explain why for me? I'd like to understand.

Comment (yaira2) [reply src/Files.App/Views/Properties/MainPropertiesPage.xaml:79]: It makes the xaml more readable

Comment (onein528) [reply src/Files.App/Views/Properties/MainPropertiesPage.xaml:79]: I definitely recommend we should not do that in DataTemplate. This cause a warning `WMC1506: OneWay bindings require at least one of their steps to support raising notifications when their value changes`.

Comment (yaira2) [reply src/Files.App/Views/Properties/MainPropertiesPage.xaml:79]: Template could be an exception for `OneWay` but I believe it's needed for `TwoWay`.

Comment (onein528) [reply src/Files.App/Views/Properties/MainPropertiesPage.xaml:79]: Yes, if a data have to be needed two way bind, have to add `Mode=TwoWay`. But should not add `Mode=OneWay` in one way binding. (In DataTemplate)

Comment (onein528) [reply src/Files.App/Views/Properties/MainPropertiesPage.xaml:79]: Default Mode is OneTime in x:Bind. OpacityIconStyle is being set in its class's constructor so I believe it's not needed. But your decision actually whether we should add OneWay for each binds in TwoWay template.

Comment (yaira2) [reply src/Files.App/Views/Properties/MainPropertiesPage.xaml:79]: I think it's better for readability and forces developers to double check that they set the correct mode.

File: src/Files.App/Views/Properties/MainPropertiesPage.xaml.cs [hunk 3]
```diff
@@ -98,6 +98,20 @@ private async void Page_Loaded(object sender, RoutedEventArgs e)
 			}
 		}
 
+		private void MainPropertiesPage_SizeChanged(object sender, SizeChangedEventArgs e)
+			=> UpdateDialogLayout();
```
Comment (yaira2) [inline src/Files.App/Views/Properties/MainPropertiesPage.xaml.cs:102]: Coding convention in Files is to use brackets

File: src/Files.App/Views/Properties/MainPropertiesPage.xaml.cs [hunk 4]
```diff
@@ -388,6 +403,13 @@ public bool IsSelected
 			get => _isSelected;
 			set => SetProperty(ref _isSelected, value);
 		}
+
+		private bool _isCompacted;
+		public bool IsCompacted
```
Comment (yaira2) [inline src/Files.App/Views/Properties/MainPropertiesPage.xaml.cs:408]: What is this property for?

File: src/Files.App/Helpers/FilePropertiesHelpers.cs [hunk 5]
```diff
@@ -58,10 +58,7 @@ public static async Task OpenPropertiesWindowAsync(object item, IShellPage assoc
 
 			if (isUniversal)
 			{
-				var frame = new Frame
-				{
-					RequestedTheme = ThemeHelper.RootTheme
-				};
+				var frame = new Frame { RequestedTheme = ThemeHelper.RootTheme };
```
Comment (yaira2) [inline src/Files.App/Helpers/FilePropertiesHelpers.cs:61]: Coding convention is not to use single line here.

File: src/Files.App/Views/Properties/MainPropertiesPage.xaml.cs (hunk 4 above)
Comment (onein528) [reply src/Files.App/Views/Properties/MainPropertiesPage.xaml.cs:408]: The labels such as General, Customisation will not be collapsed automatically when NavigationView mode is LeftCompact. To fix that, detect compact mode and collapse text with this property.

File: src/Files.App/Views/Properties/MainPropertiesPage.xaml.cs (hunk 3 above)
Comment (onein528) [reply src/Files.App/Views/Properties/MainPropertiesPage.xaml.cs:102]: Does that mean that we should not lambda expression for methods in anywhere?

File: src/Files.App/Views/Properties/MainPropertiesPage.xaml.cs (hunk 4 above)
Comment (yaira2) [reply src/Files.App/Views/Properties/MainPropertiesPage.xaml.cs:408]: Where is it set?

File: src/Files.App/Views/Properties/MainPropertiesPage.xaml.cs (hunk 3 above)
Comment (yaira2) [reply src/Files.App/Views/Properties/MainPropertiesPage.xaml.cs:102]: There might be exceptions but not in this context.

File: src/Files.App/Views/Properties/MainPropertiesPage.xaml.cs (hunk 4 above)
Comment (yaira2) [reply src/Files.App/Views/Properties/MainPropertiesPage.xaml.cs:408]: Shouldn't it be `isCompact`?

File: src/Files.App/Helpers/FilePropertiesHelpers.cs (hunk 5 above)
Comment (yaira2) [reply src/Files.App/Helpers/FilePropertiesHelpers.cs:61]: @onein528

**Review by yair100 — APPROVED** [review]

**Comment by onein528** [comment]
Oh thank you

**Pull request merged**